src/textures/CompressedArrayTexture.js

```diff
@@ -10,6 +10,14 @@ class CompressedArrayTexture extends CompressedTexture {
 		this.isCompressedArrayTexture = true;
 		this.image.depth = depth;
 		this.wrapR = ClampToEdgeWrapping;
+		this.dirtyLayers = [];
+
+	}
+
+	set layerNeedsUpdate( layerIndex ) {
```

Comment:
Since the property is called `layerNeedsUpdate` I would expect a usage like so:
```js
texture.layerNeedsUpdate = true;
```
`layerIndex` which is of type `int` is inconsistent, imo. 

How about asking devs to directly modify the `dirtyLayers` array? Similar to `drawRange` or `updateRanges` we can add similar methods for managing.

Reply:
Thinking about this: `layerNeedsUpdate` is not required at all if the feature is similar implemented to `BufferAttribute.updateRanges`. In this case, the ranges are reseted after the buffer attribute upload so we can do the same thing for layers as well.

Reply:
What would be the proposal there? To have

`CompressedArrayTextures#dirtyLayers`

in addition to

`CompressedArrayTextures#addDirtyLayer`

similar to `Buffer#updateRanges` and `Buffer#addUpdateRange`?

Reply:
Correct. Ideally, users eventually would use `addDirtyLayer()` for configuration. I would also add `clearDirtyLayers()` for resetting `dirtyLayers()` after the texture upload.

Reply:
Would it make sense to also use the `updateRanges` semantic in context of textures? Layers have a different, reserved meaning in `three.js` (see `THREE.Layers`).

Reply:
Layers also have a reserved meaning in texture arrays (hence the choice). 

https://www.khronos.org/opengl/wiki/Array_Texture

> Each mipmap level of an array texture is a series of images. Each image within a mipmap is called a "layer".

Personally, I'd rather retain the layer terminology to better align with the rendering concept so that others will have an easier time finding this feature.

Reply:
(update this PR to reflect the requested changes)